### PR TITLE
Refactoring dataset and model classes

### PR DESCRIPTION
**Pull request by SakhinetiPraveena**

### CODE REVIEW

**Comment by SakhinetiPraveena** [comment]
Thanks for the feedback! That makes sense to me as well, I'll make the changes according to that. Could you also review the functions for detection classes, anything I missed. 

And after this I am thinking defining coco.py could be a potential next step. What do you think?

**Review by dpascualhe — CHANGES_REQUESTED** [review]
I've finally had time to review the PR in depth. I've found some issues that need to be addressed, mainly:
- Fix model/dataset ontologies format mismatch (stick to `{"class_name_a": {idx: 0, rgb: [0, 0, 0]}, "class_name_b": {idx: 1, rgb: [0, 0, 0]}}, ...` format)
- Avoid doing any ontology conversion in the example notebook. The evaluated model output already matches the COCO ontology definition.
- Apply black formatter to all updated files.

Another minor thing that I'd recommend is removing the "mAP" column in the results dataframe as it is redundant. "mAP" could be reported in column "mean".

Despite these issues, the code is clear and well structured. Great job so far! Let's fix these minor issues and merge the PR 😄

PD: I've pushed to this branch https://github.com/JdeRobot/DetectionMetrics/tree/dph/object_detection the changes I've done to make the notebook example work (I'm getting a 46.6 mAP for COCO validation set, which seems reasonable)

File: detectionmetrics/datasets/coco.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        coco = COCO(annotation_file)
+    else:
+        coco = coco_obj
+    
+    # Build ontology from COCO categories
+    ontology = {}
+    for cat in coco.loadCats(coco.getCatIds()):
+        ontology[cat["name"]] = {
+            "idx": cat["id"],
+            "name": cat["name"],
```
[the remark this answers is not in the document]
Comment (SakhinetiPraveena) [reply detectionmetrics/datasets/coco.py]: Removed "name" in the ontology, in the model class to easily access class name from idx I have added a map(idx:class_name)

File: detectionmetrics/models/torch_detection.py [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+
+        dataloader = DataLoader(
+            dataset,
+            batch_size=self.model_cfg.get("batch_size", 1),
+            num_workers=self.model_cfg.get("num_workers", 1),
+            collate_fn=lambda x: tuple(zip(*x)),  # handles variable-size targets
+        )
+
+        # Init metrics
+        metrics_factory = um.DetectionMetricsFactory(iou_threshold=0.5,num_classes=self.n_classes)
```
[the remark this answers is not in the document]
Comment (SakhinetiPraveena) [reply detectionmetrics/models/torch_detection.py]: Changed this now, taking the iou_threshold from the model config. Incase it is not defined the value is defaulted to 0.5.
